lightningd/channel.c

```diff
@@ -320,6 +321,10 @@ void channel_fail_permanent(struct channel *channel, const char *fmt, ...)
 		channel->error = towire_errorfmt(channel, &cid, "%s", why);
 	}
 
+	/* Tell gossipd we no longer need to keep connection to this peer */
+	msg = towire_gossipctl_peer_important(NULL, &channel->peer->id, false);
```

Comment:
If we should consider the peer unimportant when we drop the channel to chain, would it not be better to put it in `drop_to_chain` instead of putting in two different places? That way, if we add some other condition where we drop to chain, we do not need to add this again?  The improved close of #1355 hooks into `drop_to_chain`, for example.

Reply:
Agreed, fixed.